org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/utils/NodeJsManager.java

```diff
@@ -65,6 +78,11 @@ private synchronized void initNodeIfNeeded() {
       this.nodeInit = true;
       this.nodeJsPath = helper.getNodeJsPath();
       this.nodeJsVersion = helper.getNodeJsVersion();
+      
+      if (Util.isMac()) {
+        SonarLintLogger.get().warn(
+          "Automatically detecting Node.js on macOS does not work when added to $PATH from user shell configuration!");
```

Comment:
I am not convinced this log will be really useful. Most users won't look at the console anyway. Shouldn't we log only when `nodeJsPath` is null?

Reply:
I've added a warning when it is null but kept the macOS warning. Some users might find it useful when debugging themselves. Otherwise, it will be a hint for us if we have to provide support regarding Node.js at some point 😄